bigbluebutton-html5/imports/ui/components/actions-bar/media-area/media-sharing/camera-as-content/component.tsx

```diff
@@ -471,9 +487,43 @@ const CameraAsContentView: React.FC<CameraAsContentViewProps> = ({
       hasCameraAsContent,
       stopExternalVideoShare,
       // eslint-disable-next-line no-underscore-dangle
-      true, handleFailure, { stream: PreviewService.getStream(webcamDeviceId)._mediaStream },
+      true,
+      handleFailure,
+      { stream: videoStream.mediaStream },
     );
+
     ScreenShareService.setCameraAsContentDeviceId(webcamDeviceId);
+
+    cleanupStreamAndVideo();
+
+    onActionCompleted();
+  };
+
+  useEffect(() => {
+    if (!isTransitioning) return;
+
+    if (!hasVideoStream) {
```

Comment:
Please test this with multiple cameras to see if it's working as expected. I worry this might break in the following scenario:
  - Share two cameras (regular camera sharing): c1, c2
  - Share `c1` as content

`hasVideoStream` will tentatively still be truthy because of `c2`, so the if block won't run.